File: 01_basics/01_06_gcd/gcd.js
const gcd = (a, b) => {
    if (b == 0) {
        return a;
    }

    return gcd(b, a%b);
}

const x = 4;
const y = 8;

console.log(`GCD of ${x} and ${y} is ${gcd(x, y)}`);

File: 01_basics/01_06_gcd/GCD.java
public class GCD {
    public static void main(String[] args) {
        int a = 4;
        int b = 8;

        System.out.println(String.format("GCD of %d and %d is %d", a, b, gcd(a, b)));
    }

    private static int gcd(int a, int b) {
        if ( b == 0 ) {
            return a;
        }

        return gcd(b, a%b);
    }
}
